app.py

```diff
@@ -64,23 +62,89 @@ def download_chapter(url):
     """Download a chapter."""
     return requests.get(url).content
 
+def yes_or_no(question):
+    """Prompt user for a y/n answer"""
+    while "the answer is invalid":
+        reply = str(input(question+' (y/n): ')).lower().strip()
+        if reply[:1] == 'y':
+            return True
+        if reply[:1] == 'n':
+            return False
 
 # start the app
 if __name__ == '__main__':
 
-    if len(sys.argv) != 2:
-        print(USAGE_HELP.format(*sys.argv))
-        sys.exit(1)
-
-    book_url = sys.argv[1]
-
-    with open_browser(book_url) as browser:
-        book_id = get_book_id(browser.page_source)
-        playlist = get_playist(browser, book_id)
-
-    for url, fname in playlist:
-        print('Downloading chapter "{}"'.format(fname))
-        with open('{}.mp3'.format(fname), 'wb') as outfile:
-            outfile.write(download_chapter(url))
-
-    print('All done.')
+    #set up command-line arguments
+    parser = argparse.ArgumentParser()
```

Comment:
Parser setup can be a stand-alone function (setup_parser or similar) for portability and testability sake.
You may also wish to look at [this package](https://click.palletsprojects.com) as a more elegant alternative.

Reply:
standalone function -> will do
Click -> I read about it and wanted to use it for a while, but never found a good excuse :-)
It does make the code a bit more complex but has a lot more features.
I opted for argparse, as it is the simplest form of argument parser, and is included in Python STL. However, you do require other external functions anyway, so there is no harm to add Click to the requirements as well.